include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -340,7 +340,7 @@ class operational_domain_impl
         const auto step_point_samples = generate_random_step_points(samples);
 
         // for each sample point in parallel
-        std::for_each(FICTION_EXECUTION_POLICY_PAR_UNSEQ step_point_samples.cbegin(), step_point_samples.cend(),
+        std::for_each(FICTION_EXECUTION_POLICY_SEQ step_point_samples.cbegin(), step_point_samples.cend(),
```

Comment:
@marcelwa FYI, I had to change the execution policy. Otherwise, the newly added unit test fails all the time (on windows and maybe ubuntu). ``parallel_flat_hash_map`` most likely has a problem when indentical keys (step_point_samples consists of multiple identical samples for the added test) are added/updated/accessed in parallel. It is weird, but to save time, we need to do this now.

Reply:
This is weird and really bad performance-wise. I'll try to investigate this.

Reply:
@marcelwa ah, now I see why this particular test fails. It is not caused by the ``phmap``. It is basically caused by the fact that step_point_samples can consist of ``samples`` of identical step points.

Reply:
@marcelwa Should now be fixed by ensuring the uniqueness of the sample points.